src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -4199,6 +4199,63 @@ def test_aks_skip_gpu_driver_install(self, resource_group, resource_group_locati
             checks=[self.is_empty()],
         )
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(
+        random_name_length=17,
+        name_prefix="clitest",
+        location="eastus2",
+    )
+    def test_aks_gpu_driver_type(self, resource_group, resource_group_location):
+        print(resource_group_location)
+        create_version, upgrade_version = self._get_versions(resource_group_location)
+        aks_name = self.create_random_name("cliakstest", 16)
+        nodepool_name = self.create_random_name("c", 6)
+
+        self.kwargs.update(
+            {
+                "resource_group": resource_group,
+                "name": aks_name,
+                "location": resource_group_location,
+                "nodepool_name": nodepool_name,
+                "k8s_version": upgrade_version,
+                "ssh_key_value": self.generate_ssh_keys(),
+                "windows_admin_username": "azureuser1",
+                "windows_admin_password": "replace-Password1234$",
+            }
+        )
+
+        # create an aks cluster
+        create_cmd = (
+            "aks create --resource-group {resource_group} --name {name} --location {location} "
+            "--node-count 2 "
+            "--windows-admin-username={windows_admin_username} --windows-admin-password={windows_admin_password} "
+            "--load-balancer-sku=standard --vm-set-type=virtualmachinescalesets --network-plugin=azure "
+            "-k {k8s_version} "
+            "--ssh-key-value={ssh_key_value} -o json"
+        )
+        self.cmd(
+            create_cmd, checks=[self.check("provisioningState", "Succeeded")]
+        )
+
+        # create nodepool from the cluster with custom driver type GRID
+        create_nodepool_cmd = (
+            "aks nodepool add --resource-group={resource_group} --cluster-name={name} --name={nodepool_name} --os-type windows --node-count 1 "
+            "--node-vm-size Standard_NC4as_T4_v3 --driver-type GRID "
+            "-k {k8s_version} -o json"
+        )
+               
+        self.cmd(
+            create_nodepool_cmd,
+            checks=[self.check("provisioningState", "Succeeded"),
+                    self.check('gpuProfile.driverType', "GRID")],
+        )
+
```

Comment:
also update this node pool (az aks nodepool update ...), and validate that the driver type should not change

Reply:
I tried adding this scenario to the live test but I got some errors when I run the test because the `--driver-type` is not added to the `az aks nodepool update` command.

<img width="751" alt="image" src="https://github.com/user-attachments/assets/19283b84-0fbe-4859-b90f-c83c776f5189">

 Any update to driver-type is blocked by the RP FE logic. (similar to --skip-gpu-driver-install) driverType is an immutable property upon nodepool creation.

Reply:
yes, 'az aks nodepool update' doesn't have --driver-type, I just want to see that 'az aks nodepool update' doesn't change the driver type and doesn't fail.

Reply:
will add this scenario in a separate PR